RInAppMessaging/Classes/Repositories/CampaignRepository.swift

```diff
@@ -129,7 +143,16 @@ internal class CampaignRepository: CampaignRepositoryType {
         let updatedCampaign = Campaign.updatedCampaign(campaign, withImpressionLeft: newValue)
         list[index] = updatedCampaign
         campaigns.set(value: list)
-        userDataCache.cacheCampaignData(list, userIdentifiers: preferenceRepository.getUserIdentifiers())
+        saveDataToCache(list)
+
         return updatedCampaign
     }
+
+    private func saveDataToCache(_ list: [Campaign]) {
+        let user = preferenceRepository.getUserIdentifiers()
+        userDataCache.cacheCampaignData(list, userIdentifiers: user)
+        if defaultCacheUser != user {
+            userDataCache.cacheCampaignData(list, userIdentifiers: defaultCacheUser)
```

Comment:
Why do we need this?

Reply:
Any update in logged-in user cache is also reflected in anon/default user cache so the same data can be picked up at the next app launch. Keep in mind that there's ping call in between which additionally ensures that campaign list is up to date